matrix-sdk/src/main/java/org/matrix/androidsdk/ssl/CertUtil.java

```diff
@@ -0,0 +1,58 @@
+package org.matrix.androidsdk.ssl;
+
+import java.security.MessageDigest;
+import java.security.cert.X509Certificate;
+
+/**
+ * Various utility classes for dealing with X509Certificates
+ */
+public class CertUtil {
+
+    /**
+     * Generates the SHA-256 fingerprint of the given certificate
+     * @param cert
+     * @return
+     */
+    public static byte[] generateSha256Fingerprint(X509Certificate cert) {
+        final byte[] fingerprint;
+        try {
+            MessageDigest md = MessageDigest.getInstance("SHA256");
+            fingerprint = md.digest(cert.getEncoded());
+        } catch(Exception e) {
+            throw new RuntimeException(e);
```

Comment:
Is a `RuntimeException` really appropriate here? This isn't recoverable in any way? (e.g. by rejecting the cert?)